tests/components/cpuspeed/test_config_flow.py

```diff
@@ -74,3 +79,31 @@ async def test_import_flow(
     assert result.get("data") == {}
 
     assert len(mock_setup_entry.mock_calls) == 1
+    assert len(mock_cpuinfo_config_flow.mock_calls) == 1
+
+
+async def test_not_compatible(
+    hass: HomeAssistant,
+    mock_cpuinfo_config_flow: MagicMock,
+    mock_setup_entry: AsyncMock,
+) -> None:
+    """Test we abort the configuration flow when incompatible."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    assert result.get("type") == RESULT_TYPE_FORM
```

Comment:
```suggestion
    assert result["type"] == RESULT_TYPE_FORM
```

Is the result type ever missing?

Reply:
Yes, according to the typing it is.

Reply:
Looks like we have it both ways in config flow tests 🤷‍♂️

Reply:
Yeah, we do indeed. I generally code/check for typing in tests as well; but that is more my personal setup and slightly drive to have no red stuff in my screen polluting the code (my brain can't handle that 😓 )